### ISSUE TITLE
After worker nodes boot it takes several minutes to get registered as nodes

### ISSUE THREAD

**Issue by mcornea** [openshift-metal3/dev-scripts#570]
**Describe the bug**
After worker nodes boot it takes several minutes to get registered as nodes.

Going through the kubelet log of a worker node we can see it first tries to register the node at 15:15:10:

```sh
[root@rhhi-node-3 core]# journalctl -l -u kubelet | grep 'register node' | head -1
May 24 15:15:10 rhhi-node-3 hyperkube[1240]: I0524 15:15:10.143247    1240 kubelet_node_status.go:72] Attempting to register node rhhi-node-3
```

but it only manages to register the node after 8 minutes.

```sh
[root@rhhi-node-3 core]# journalctl -l -u kubelet | grep 'registered node'
May 24 15:23:40 rhhi-node-3 hyperkube[1240]: I0524 15:23:40.111852    1240 kubelet_node_status.go:75] Successfully registered node rhhi-node-3
``` 

Based on my observations this time can vary, the longest that I've seen was around 15 minutes.

The kubelet log shows the following errors when the node doesn't get registered:

```sh
May 24 15:23:33 rhhi-node-3 hyperkube[1240]: E0524 15:23:33.094148    1240 kubelet_node_status.go:94] Unable to register node "rhhi-node-3" with API server: nodes is forbidden: User "system:anonymous" cannot create resource "nodes" in API group "" at the cluster scope
```

**To Reproduce**
Deploy worker nodes and watch the kubelet log on the worker node after it boots.

**Expected/observed behavior**

It takes several minutes for the worker node to get registered. I'd expect that the worker node would get registered right away when kubelet starts.

**Additional context**
kubelet.log - https://paste.fedoraproject.org/paste/FadFlN9S8JLD0m94YGM6qg

**Comment by vrutkovs** [comment]
>I'd expect that the worker node would get registered right away when kubelet starts.

The node got registered when SDN container started, which requires a few images to be pulled, have mdns and coredns services to start.

**Comment by mcornea** [comment]
> > I'd expect that the worker node would get registered right away when kubelet starts.
> 
> The node got registered when SDN container started, which requires a few images to be pulled, have mdns and coredns services to start.

It looks like the mdns and containers get started ~ 15:15 but the sdn container gets started ~ 15:23 which corresponds to the time when the node gets registered. So I guess the time to register varies based on the link parameters to quay.io where these images are hosted.

```sh
[root@rhhi-node-3 core]# for container in $(crictl ps | grep -v NAME | awk {'print $8'}); do echo $container; crictl inspect $(crictl ps | grep $container | awk {'print $1'}) | grep startedAt;done
node-ca
    "startedAt": "2019-05-24T15:24:32.040460121Z",
machine-config-daemon
    "startedAt": "2019-05-24T15:24:13.026092833Z",
dns-node-resolver
    "startedAt": "2019-05-24T15:24:10.386422198Z",
dns
    "startedAt": "2019-05-24T15:24:10.386422198Z",
    "startedAt": "2019-05-24T15:24:06.95502962Z",
    "startedAt": "2019-05-24T15:15:52.159969349Z",
    "startedAt": "2019-05-24T15:15:45.606445636Z",
kube-multus
    "startedAt": "2019-05-24T15:24:06.938282616Z",
openvswitch
    "startedAt": "2019-05-24T15:23:58.791896636Z",
kube-rbac-proxy
    "startedAt": "2019-05-24T15:23:58.779890312Z",
sdn
    "startedAt": "2019-05-24T15:23:58.744983994Z",
tuned
    "startedAt": "2019-05-24T15:23:52.569024449Z",
node-exporter
    "startedAt": "2019-05-24T15:23:49.827651572Z",
mdns-publisher
    "startedAt": "2019-05-24T15:15:52.159969349Z",
coredns
    "startedAt": "2019-05-24T15:15:45.606445636Z",

```

**Comment by hardys** [comment]
I'm seeing similar, the time for the workers to deploy and register is about 35mins, which I'm pretty sure is longer than we were previously seeing:

```
$ oc get nodes
NAME       STATUS   ROLES    AGE   VERSION
master-0   Ready    master   45m   v1.13.4+cb455d664
master-1   Ready    master   45m   v1.13.4+cb455d664
master-2   Ready    master   46m   v1.13.4+cb455d664
worker-0   Ready    worker   10m   v1.13.4+cb455d664
worker-1   Ready    worker   10m   v1.13.4+cb455d664
worker-2   Ready    worker   10m   v1.13.4+cb455d664

```

**Comment by russellb** [comment]
I suspect this is the cert approval issue, #260 

Issue #597 looks like the same root cause, as well.

Short answer: run fix_certs.sh.

Patch to mitigate this quickly: shorten the interval we run fix_certs.sh.  We should just run it every minute or something instead of 30m.

#260 tracks fixing all of this properly.

We have some automation for the default single worker dev env case that makes it go quickly.  Otherwise, you have to run link-machine-and-node.sh manually or wait for the timer that runs fix_certs.

**Comment by hardys** [comment]
Ok lets close this, we've landed the short-term workaround, and #260 tracks the long-term fix
